src/Compiler.jl

```diff
@@ -625,6 +625,8 @@ function optimization_passes(; no_nan::Bool=false, sroa::Bool=false, inline::Boo
             passes,
             "sroa-wrappers{instcombine=false instsimplify=true $(SROA_ATTRIBUTOR[] ? "" : "attributor=false")}",
         )
+
+        push!(passes, "canonicalize")
```

Comment:
Can you try to run lib device raise twice like this ?

```suggestion
        push!(passes, "libdevice-funcs-raise")
        push!(passes, "canonicalize")
```

Reply:
actually im curious if as is fixes it